tools/twix/src/panels/map/layers/feet_detection.rs

```diff
@@ -0,0 +1,90 @@
+
+use color_eyre::Result;
+use communication::client::{Cycler, CyclerOutput, Output};
+use eframe::epaint::Color32;
+use nalgebra::{Isometry2, Point2};
+
+use types::{ClusterPoint, FieldDimensions};
+
+use crate::{
+    nao::Nao, panels::map::layer::Layer, twix_painter::TwixPainter, value_buffer::ValueBuffer,
+};
+
+pub struct FeetDetection {
+    robot_to_field: ValueBuffer,
+    cluster_bottom: ValueBuffer,
+    cluster_top: ValueBuffer,
+    segments_bottom: ValueBuffer,
+    segments_top: ValueBuffer,
+}
+
+impl Layer for FeetDetection {
+    const NAME: &'static str = "FeetDetection";
+
+    fn new(nao: Arc<Nao>) -> Self {
+        let robot_to_field = nao.subscribe_output(CyclerOutput {
+            cycler: Cycler::Control,
+            output: Output::Main {
+                path: "robot_to_field".to_string(),
+            },
+        });
+        let cluster_bottom = nao.subscribe_output(CyclerOutput {
+            cycler: Cycler::VisionBottom,
+            output: Output::Additional {
+                path: "feet_detection.clusters_in_ground".to_string(),
+            },
+        });
+        let cluster_top = nao.subscribe_output(CyclerOutput {
+            cycler: Cycler::VisionTop,
+            output: Output::Additional {
+                path: "feet_detection.clusters_in_ground".to_string(),
+            },
+        });
+        let segments_bottom = nao.subscribe_output(CyclerOutput {
+            cycler: Cycler::VisionBottom,
+            output: Output::Additional {
+                path: "feet_detection.cluster_points".to_string(),
+            },
+        });
+        let segments_top = nao.subscribe_output(CyclerOutput {
+            cycler: Cycler::VisionTop,
+            output: Output::Additional {
+                path: "feet_detection.cluster_points".to_string(),
+            },
+        });
+        Self {
+            robot_to_field,
+            cluster_bottom,
+            cluster_top,
+            segments_bottom,
+            segments_top,
+        }
+    }
+
+    fn paint(&self, painter: &TwixPainter, _field_dimensions: &FieldDimensions) -> Result<()> {
+        let robot_to_field: Isometry2<f32> = self.robot_to_field.parse_latest().unwrap_or_default();
+        let cluster_points: Vec<Point2<f32>> = [&self.cluster_bottom, &self.cluster_top]
+            .iter()
+            .filter_map(|buffer| buffer.parse_latest::<Vec<_>>().ok())
+            .flatten()
+            .collect();
+        for point in cluster_points {
+            painter.circle_filled(robot_to_field * point, 0.05, Color32::YELLOW);
+        }
+
+        let cluster_points: Vec<ClusterPoint> = [&self.segments_bottom, &self.segments_top]
+            .iter()
+            .filter_map(|buffer| buffer.parse_latest::<Vec<_>>().ok())
+            .flatten()
+            .collect();
+        for point in cluster_points {
+            painter.circle_filled(
+                robot_to_field * point.position_in_ground,
+                0.02,
```

Comment:
Maybe consider to use a constant to describe the radius.

Reply:
I extracted the `0.02` to a variable

Reply:
Would't a constant be more appropriate?

Reply:
I think the best would be to have it configurable, but that is definitely future work. This is debug tooling and a constant moves the information far away from this specific implementation. Additionally, I think introducing a constant mainly makes sense if it is used multiple times. I propose to keep it like this.